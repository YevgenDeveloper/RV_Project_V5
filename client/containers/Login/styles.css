@import 'client/styles/variables';

.container {
  position: relative;
  width: 100%;
  display: flex;

  &::before {
    z-index: -1;
    content: '';
    position: fixed;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    background: url('client/assets/bg.jpg') 50% 50% no-repeat;
    background-size: cover;
    animation: bg-in 1.5s cubic-bezier(0.165, 0.84, 0.44, 1);
    transition-property: opacity, transform;
    transition-duration: 1.5s;
    transition-timing-function: cubic-bezier(0.95, 0.05, 0.795, 0.035);
  }

  &.request::before {
    opacity: 0;
    transform: scale(1.1);
  }
}

.inner {
  margin: auto;
  width: 240px;
  animation: inner-in 1.5s cubic-bezier(0.165, 0.84, 0.44, 1);
  transition-property: opacity, transform;
  transition-duration: 1.5s;
  transition-timing-function: cubic-bezier(0.95, 0.05, 0.795, 0.035);

  .request & {
    opacity: 0;
    transform: scale(1.2);
  }
}

.buttonSumit {
  width: 100%;
  height: 40px;
}

.footer {
  margin: 40px auto;
  color: $color-text-default;
  text-align: center;
  line-height: 1.6em;
}

.logoSeeItAll {
  display: block;
  margin: 5px auto;
  overflow: hidden;
  text-indent: -1000px;
  width: 80px;
  height: 30px;
  background: url('client/assets/logo-seeitall.png') 50% 50% no-repeat;
  background-size: 80px 30px;
}

@keyframes bg-in {
  0% {
    opacity: 0;
    transform: scale(1.1);
  }

  100% {
    opacity: 1;
    transform: scale(1);
  }
}

@keyframes inner-in {
  0% {
    opacity: 0;
    transform: scale(1.2);
  }

  100% {
    opacity: 1;
    transform: scale(1);
  }
}
