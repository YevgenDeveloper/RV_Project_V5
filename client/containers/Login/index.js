// @flow

import React, { Component } from 'react';
import classNames from 'classnames/bind';

import { Button, Input, Logo } from 'client/components';
import { login } from 'client/services';
import styles from './styles.css';

const YEAR = new Date().getFullYear();
const VERSION = '0.0.1';

const cx = classNames.bind(styles);

type State = {
  email: string,
  password: string,
  isRequest: boolean,
  errorMessage: string,
};

export default class Login extends Component<{}, State> {
  timeout = null;

  state = {
    email: '',
    password: '',
    isRequest: true,
    errorMessage: '',
  };

  componentDidMount() {
    this.timeout = setTimeout(() => this.setState({ isRequest: false }), 1500);
  }

  componentWillUnmount() {
    clearTimeout(this.timeout);
  }

  handleRequest = () => {
    this.setState({ isRequest: true });

    login();

    this.timeout = setTimeout(() => window.location.reload(), 1500);
  };

  timeout: TimeoutID;

  render() {
    const { isRequest, errorMessage } = this.state;

    return (
      <div className={cx(styles.container, { [styles.request]: isRequest })}>
        <div className={styles.inner}>
          <Logo icon />
          <Input
            type="email"
            value={this.state.email}
            placeholder="Enter email"
            onChange={(email: string) => this.setState({ email })}
            disabled={isRequest}
          />
          <Input
            type="password"
            value={this.state.password}
            placeholder="Enter password"
            onChange={(password: string) => this.setState({ password })}
            disabled={isRequest}
          />
          {errorMessage && <div className={styles.error}>{errorMessage}</div>}
          <Button
            blue
            label="Submit"
            className={styles.buttonSumit}
            onClick={this.handleRequest}
            disabled={isRequest}
          />
          <div className={styles.footer}>
            {`Â© ${YEAR} RailView ver. ${VERSION}`}
            <br />
            Powered by
            <a
              href="http://seeitall.nl/"
              className={styles.logoSeeItAll}
              rel="noopener noreferrer"
              target="_blank"
            >
              See IT all
            </a>
          </div>
        </div>
      </div>
    );
  }
}
