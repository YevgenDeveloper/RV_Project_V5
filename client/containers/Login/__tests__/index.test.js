// @flow

import type { ShallowWrapper } from 'enzyme';

import React from 'react';
import { shallow } from 'enzyme';
import Login from '..';

it('renders correctly', () => {
  const wrapper: ShallowWrapper = shallow(<Login />);

  expect(wrapper).toMatchSnapshot();
});
