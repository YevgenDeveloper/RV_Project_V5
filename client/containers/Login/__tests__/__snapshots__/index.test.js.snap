// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="inner"
      >
        <Logo
          icon={true}
        />
        <Input
          disabled={true}
          onChange={[Function]}
          placeholder="Enter email"
          type="email"
          value=""
        />
        <Input
          disabled={true}
          onChange={[Function]}
          placeholder="Enter password"
          type="password"
          value=""
        />
        
        <Button
          blue={true}
          className="buttonSumit"
          dark={false}
          disabled={true}
          label="Submit"
          onClick={[Function]}
        />
        <div
          className="footer"
        >
          © 2018 RailView ver. 0.0.1
          <br />
          Powered by
          <a
            className="logoSeeItAll"
            href="http://seeitall.nl/"
            rel="noopener noreferrer"
            target="_blank"
          >
            See IT all
          </a>
        </div>
      </div>,
      "className": "container request",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Logo
            icon={true}
          />,
          <Input
            disabled={true}
            onChange={[Function]}
            placeholder="Enter email"
            type="email"
            value=""
          />,
          <Input
            disabled={true}
            onChange={[Function]}
            placeholder="Enter password"
            type="password"
            value=""
          />,
          "",
          <Button
            blue={true}
            className="buttonSumit"
            dark={false}
            disabled={true}
            label="Submit"
            onClick={[Function]}
          />,
          <div
            className="footer"
          >
            © 2018 RailView ver. 0.0.1
            <br />
            Powered by
            <a
              className="logoSeeItAll"
              href="http://seeitall.nl/"
              rel="noopener noreferrer"
              target="_blank"
            >
              See IT all
            </a>
          </div>,
        ],
        "className": "inner",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": undefined,
            "icon": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": undefined,
            "disabled": true,
            "onChange": [Function],
            "placeholder": "Enter email",
            "type": "email",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": undefined,
            "disabled": true,
            "onChange": [Function],
            "placeholder": "Enter password",
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "blue": true,
            "className": "buttonSumit",
            "dark": false,
            "disabled": true,
            "label": "Submit",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "© 2018 RailView ver. 0.0.1",
              <br />,
              "Powered by",
              <a
                className="logoSeeItAll"
                href="http://seeitall.nl/"
                rel="noopener noreferrer"
                target="_blank"
              >
                See IT all
              </a>,
            ],
            "className": "footer",
          },
          "ref": null,
          "rendered": Array [
            "© 2018 RailView ver. 0.0.1",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            "Powered by",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "See IT all",
                "className": "logoSeeItAll",
                "href": "http://seeitall.nl/",
                "rel": "noopener noreferrer",
                "target": "_blank",
              },
              "ref": null,
              "rendered": "See IT all",
              "type": "a",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="inner"
        >
          <Logo
            icon={true}
          />
          <Input
            disabled={true}
            onChange={[Function]}
            placeholder="Enter email"
            type="email"
            value=""
          />
          <Input
            disabled={true}
            onChange={[Function]}
            placeholder="Enter password"
            type="password"
            value=""
          />
          
          <Button
            blue={true}
            className="buttonSumit"
            dark={false}
            disabled={true}
            label="Submit"
            onClick={[Function]}
          />
          <div
            className="footer"
          >
            © 2018 RailView ver. 0.0.1
            <br />
            Powered by
            <a
              className="logoSeeItAll"
              href="http://seeitall.nl/"
              rel="noopener noreferrer"
              target="_blank"
            >
              See IT all
            </a>
          </div>
        </div>,
        "className": "container request",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Logo
              icon={true}
            />,
            <Input
              disabled={true}
              onChange={[Function]}
              placeholder="Enter email"
              type="email"
              value=""
            />,
            <Input
              disabled={true}
              onChange={[Function]}
              placeholder="Enter password"
              type="password"
              value=""
            />,
            "",
            <Button
              blue={true}
              className="buttonSumit"
              dark={false}
              disabled={true}
              label="Submit"
              onClick={[Function]}
            />,
            <div
              className="footer"
            >
              © 2018 RailView ver. 0.0.1
              <br />
              Powered by
              <a
                className="logoSeeItAll"
                href="http://seeitall.nl/"
                rel="noopener noreferrer"
                target="_blank"
              >
                See IT all
              </a>
            </div>,
          ],
          "className": "inner",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": undefined,
              "icon": true,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": undefined,
              "disabled": true,
              "onChange": [Function],
              "placeholder": "Enter email",
              "type": "email",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": undefined,
              "disabled": true,
              "onChange": [Function],
              "placeholder": "Enter password",
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "blue": true,
              "className": "buttonSumit",
              "dark": false,
              "disabled": true,
              "label": "Submit",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "© 2018 RailView ver. 0.0.1",
                <br />,
                "Powered by",
                <a
                  className="logoSeeItAll"
                  href="http://seeitall.nl/"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  See IT all
                </a>,
              ],
              "className": "footer",
            },
            "ref": null,
            "rendered": Array [
              "© 2018 RailView ver. 0.0.1",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              "Powered by",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "See IT all",
                  "className": "logoSeeItAll",
                  "href": "http://seeitall.nl/",
                  "rel": "noopener noreferrer",
                  "target": "_blank",
                },
                "ref": null,
                "rendered": "See IT all",
                "type": "a",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
